From 8b761fd757bad28531f3192f6097576a460bc3fb Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 29 Dec 2022 06:27:08 +0200
Subject: [PATCH] [SW-113135] gaudi3: use reset source enum from specs

Currently, ppboot and preboot has different enums defined for
reset source. One is defined with values incremented with bit shifts
and other defined as bit numbers. Since the value read from BTM FIFOs
reflects bit numbers, this patch removes reset source enum defined in
ppboot and use it from specs file.

Change-Id: I5ffe8c7ccb2659d9e47ae68fe99d03b5e798f594
---

diff --git a/arch/gaudi3/arc/hl.h b/arch/gaudi3/arc/hl.h
index fe87240..85c5ef7 100644
--- a/arch/gaudi3/arc/hl.h
+++ b/arch/gaudi3/arc/hl.h
@@ -10,6 +10,7 @@
 #include <preboot.h>
 #include <d0_parc_global_conf_regs.h>
 #include <d0_parc_global_conf_masks.h>
+#include <gaudi3_fw_if.h>
 
 #define BOOT_DESC_MAX_IMGS	7
 #define READ_BOOT_DESC_FROM_BOOTINFO
@@ -36,18 +37,6 @@
 #define HL_PLL_DIV_FACTOR_CMD	0x40
 #define HL_PLL_DIV_SEL		0x50
 #define HL_PLL_DIV_EN		0x60
-
-enum gaudi3_rst_src {
-	HL_COLD_RST = 1,
-	HL_MANUAL_RST = 2,
-	HL_PRSTN_RST = 4,
-	HL_SOFT_RST = 8,
-	HL_WD_RST = 16,
-	HL_FW_ALL_RST = 32,
-	HL_SW_ALL_RST = 64,
-	HL_FLR_RST = 128,
-	HL_ECC_DERR_RST = 256
-};
 
 enum gaudi3_die_id {
 	GAUDI3_MASTER_DIE_ID,
